components/locale_core/src/preferences/locale.rs

```diff
@@ -61,25 +61,23 @@ impl Default for LocalePreferences {
 
 impl From<&crate::Locale> for LocalePreferences {
     fn from(loc: &crate::Locale) -> Self {
-        let sd = loc
-            .extensions
-            .unicode
-            .keywords
-            .get(&crate::extensions::unicode::key!("sd"))
-            .and_then(|v| SubdivisionId::try_from_str(v.as_single_subtag()?.as_str()).ok());
-        let ue_region = loc
-            .extensions
-            .unicode
-            .keywords
-            .get(&crate::extensions::unicode::key!("rg"))
-            .and_then(|v| SubdivisionId::try_from_str(v.as_single_subtag()?.as_str()).ok());
         Self {
             language: loc.id.language,
             script: loc.id.script,
             region: loc.id.region,
             variant: loc.id.variants.iter().copied().next(),
-            subdivision: sd,
-            ue_region,
+            subdivision: loc
+                .extensions
+                .unicode
+                .keywords
+                .get(&RegionalSubdivision::UNICODE_EXTENSION_KEY)
+                .and_then(|v| RegionalSubdivision::try_from(v.clone()).ok()),
```

Comment:
Issue: I'm not sure if the compiler is smart enough to realize that the `.clone()` doesn't do anything. It is not a trivial clone impl because `Value` has a vec inside. This TryFrom impl should be on `&Value` not `Value`.

OK for an immediate follow-up if you prefer